src/status_im/wallet/choose_recipient/core.cljs

```diff
@@ -84,9 +84,17 @@
         (if (:wallet/prepare-transaction db)
           {:db (update db :wallet/prepare-transaction assoc
                        :to address :to-name (find-address-name db address))}
-          (let [current-chain-id (get-in networks [current-network :config :NetworkId])]
-            (merge {:db (fill-prepare-transaction-details db details all-tokens)
+          (let [current-chain-id                     (get-in networks [current-network :config :NetworkId])
+                new-db-with-transaction-from-details (fill-prepare-transaction-details db details all-tokens)]
+            (merge {:db       new-db-with-transaction-from-details
                     :dispatch [:open-modal :prepare-send-transaction]}
+                   (when-not (get-in new-db-with-transaction-from-details [:wallet/prepare-transaction :gasPrice])
```

Comment:
i guess you could just check it in details ?

Reply:
```
[{:keys [address gasPrice] :as details}
        (eip681/extract-request-details data all-tokens)]
        ```

Reply:
fixed